kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/resolver/DependencyResolverImplTest.java

```diff
@@ -63,226 +78,658 @@ void testNoDependencies() {
         assertThat(resolutionResult.filters()).isEmpty();
     }
 
+    @Test
+    void deepResolveClusterNoDependenciesVirtualKafkaCluster() {
+        givenFiltersInContext();
```

Comment:
How about making setup establish the context returning no filters, no cluster refs, and leaving the test to set just what it needs.